### DISCUSSION TITLE
Unset ARGs in FROM statement leads to "invalid-name"

### DISCUSSION THREAD

**Discussion by idelchi** [renovatebot/renovate#31010] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, v37.440.7

### Please tell us more about your question or problem

A Dockerfile with an ARG declaration for the registry source

```
ARG REGISTRY
FROM ${REGISTRY}python:3.12-slim
```

leads to REGISTRY not being resolved correctly (see logs)

whereas 

```
ARG REGISTRY=
FROM ${REGISTRY}python:3.12-slim
```

will correctly scan dependencies from `docker.io`.

Is there a way to achieve the same behaviour without needing to set the ARG to an empty value?

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: packageFiles with updates (repository=local)
       "config": {
         "dockerfile": [
           {
             "deps": [
               {
                 "skipReason": "invalid-name",
                 "replaceString": "${REGISTRY}python:3.12-slim",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "updates": []
               }
             ],
             "packageFile": "images\\cloud\\Dockerfile"
           }
         ]
       }
```

</details>

**Comment by rarkins** [comment, marked as answer]
Try registryAliases

**Reply by idelchi** [reply]
Thank you @rarkins !
